api/repository.go

```diff
@@ -107,14 +110,15 @@ type RepositoryPatch struct {
 	UpdaterID int
 
 	// Domain specific fields
-	BranchFilter       *string `jsonapi:"attr,branchFilter"`
-	BaseDirectory      *string `jsonapi:"attr,baseDirectory"`
-	FilePathTemplate   *string `jsonapi:"attr,filePathTemplate"`
-	SchemaPathTemplate *string `jsonapi:"attr,schemaPathTemplate"`
-	SheetPathTemplate  *string `jsonapi:"attr,sheetPathTemplate"`
-	AccessToken        *string
-	ExpiresTs          *int64
-	RefreshToken       *string
+	BranchFilter        *string `jsonapi:"attr,branchFilter"`
+	BaseDirectory       *string `jsonapi:"attr,baseDirectory"`
+	FilePathTemplate    *string `jsonapi:"attr,filePathTemplate"`
+	SchemaMigrationType *string `jsonapi:"attr,schemaMigrationType"`
```

Comment:
Applying the following would not work with error:

```
Can't unmarshal DDL (string) to struct field `SchemaMigrationType`, which is a pointer to `RepositorySchemaMigrationType (string)`
```

```suggestion
	SchemaMigrationType * ProjectSchemaMigrationType `jsonapi:"attr,schemaMigrationType"`
```

Which means the existing but unused code would also not going to work?

https://github.com/bytebase/bytebase/blob/77b2f39fd53352230aeaf4fa0c2051fc5cf934c1/api/project.go#L135

Reply:
@d-bytebase do you have an idea of where the problem is?

Reply:
Boom, that's an issue with google/jsonapi... Lots of interface and reflection. We have to use string to workaround the problem.